File: UdpClient.cpp
#include <iostream>
#include <cstring>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <unistd.h>
#define PORT 8080

int main() {
    int sockfd;
    char buffer[1024];
    const char *message = "Hello from UDP Client";
    struct sockaddr_in servaddr;

    // Create socket
    if ((sockfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
        perror("Socket creation failed");
        return -1;
    }

    // Address configuration
    memset(&servaddr, 0, sizeof(servaddr));
    servaddr.sin_family = AF_INET;
    servaddr.sin_port = htons(PORT);
    servaddr.sin_addr.s_addr = INADDR_ANY;

    // Send message to server
    sendto(sockfd, message, strlen(message), MSG_CONFIRM, (const struct sockaddr *)&servaddr, sizeof(servaddr));
    std::cout << "Message sent to server\n";

    // Receive response from server
    int len, n;
    len = sizeof(servaddr);
    n = recvfrom(sockfd, buffer, 1024, MSG_WAITALL, (struct sockaddr *)&servaddr, (socklen_t *)&len);
    buffer[n] = '\0';
    std::cout << "Message from server: " << buffer << "\n";

    // Close socket
    close(sockfd);
    return 0;
}

File: UDPClient.java
import java.net.*;

public class UDPClient {
    public static void main(String[] args) {
        String serverAddress = "127.0.0.1";
        int port = 8080;
        String message = "Hello from UDP Client";
        
        try (DatagramSocket socket = new DatagramSocket()) {
            byte[] buffer = message.getBytes();
            
            InetAddress serverIP = InetAddress.getByName(serverAddress);
            DatagramPacket packet = new DatagramPacket(buffer, buffer.length, serverIP, port);
            socket.send(packet);
            
            byte[] responseBuffer = new byte[1024];
            DatagramPacket responsePacket = new DatagramPacket(responseBuffer, responseBuffer.length);
            socket.receive(responsePacket);
            
            String response = new String(responsePacket.getData(), 0, responsePacket.getLength());
            System.out.println("Message from server: " + response);
        } catch (IOException ex) {
            ex.printStackTrace();
        }
    }
}
